audformat/core/utils.py

```diff
@@ -1139,9 +1139,9 @@ def set_index_dtypes(
 
     if isinstance(index, pd.MultiIndex):
         # MultiIndex
-        if all([len(level) == 0 for level in index.levels]):
-            # set_levels() does not work
-            # in the case the levels are something like `[[], []]`,
+        if any([len(index.levels[index.names.index(level)]) == 0
+                for level in dtypes]):
```

Comment:
Maybe slightly easier:

```suggestion
        if any([len(level) == 0 for level in index.levels]):
```

Reply:
Then we enter the `if` clause always when there an empty level even if we actually won't change it.

Reply:
Ah, ok, now I get it.